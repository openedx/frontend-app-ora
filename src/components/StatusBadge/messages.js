import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  inProgressBadge: {
    id: 'frontend-app-ora.StatusAlert.inProgressBadge',
    defaultMessage: 'In Progress',
    description: 'Label for the in progress badge',
  },
  completedBadge: {
    id: 'frontend-app-ora.StatusAlert.completedBadge',
    defaultMessage: 'Completed',
    description: 'Label for the completed badge',
  },
  errorBadge: {
    id: 'frontend-app-ora.StatusAlert.errorBadge',
    defaultMessage: 'Incomplete',
    description: 'Label for the incomplete badge',
  },
  cancelledBadge: {
    id: 'frontend-app-ora.StatusAlert.cancelledBadge',
    defaultMessage: 'Cancelled',
    description: 'Label for the cancelled badge',
  },
});

export default messages;
