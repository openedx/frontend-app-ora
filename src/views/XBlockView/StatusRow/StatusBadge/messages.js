import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  cancelled: {
    defaultMessage: 'Cancelled',
    description: 'Cancelled badge message',
    id: 'frontend-app-ora.XBlockView.StatusRow.cancelled',
  },
  notReady: {
    defaultMessage: 'Not ready yet',
    description: 'Not ready yet badge message',
    id: 'frontend-app-ora.XBlockView.StatusRow.notReady',
  },
  waiting: {
    defaultMessage: 'Waiting',
    description: 'Waiting badge message',
    id: 'frontend-app-ora.XBlockView.StatusRow.waiting',
  },
  notAvailable: {
    defaultMessage: 'Not available yet',
    description: 'Not available yet badge message',
    id: 'frontend-app-ora.XBlockView.StatusRow.notAvailable',
  },
  pastDue: {
    defaultMessage: 'Past due',
    description: 'Past-due badge message',
    id: 'frontend-app-ora.XBlockView.StatusRow.pastDue',
  },
  closed: {
    defaultMessage: 'Closed',
    description: 'Closed badge message',
    id: 'frontend-app-ora.XBlockView.StatusRow.closed',
  },
  teamRequired: {
    defaultMessage: 'Team required',
    description: 'Team-required badge message',
    id: 'frontend-app-ora.XBlockView.StatusRow.teamRequired',
  },
  inProgress: {
    defaultMessage: 'In progress',
    description: 'In-progress badge message',
    id: 'frontend-app-ora.XBlockView.StatusRow.inProgress',
  },
  submitted: {
    defaultMessage: 'Response submitted',
    description: 'Response submitted badge message',
    id: 'frontend-app-ora.XBlockView.StatusRow.submitted',
  },
  practiceCompleted: {
    defaultMessage: 'Practice grading is completed',
    description: 'Practice grading complete badge message',
    id: 'frontend-app-ora.XBlockView.StatusRow.practiceCompleted',
  },
  selfCompleted: {
    defaultMessage: 'Self assessment is completed',
    description: 'Self assessment complete badge message',
    id: 'frontend-app-ora.XBlockView.StatusRow.selfCompleted',
  },
  peerCompleted: {
    defaultMessage: 'Peer assessment is completed',
    description: 'Peer assessment complete badge message',
    id: 'frontend-app-ora.XBlockView.StatusRow.peerCompleted',
  },
  complete: {
    defaultMessage: 'Complete',
    description: 'Complete badge message',
    id: 'frontend-app-ora.XBlockView.StatusRow.complete',
  },
});

export default messages;
