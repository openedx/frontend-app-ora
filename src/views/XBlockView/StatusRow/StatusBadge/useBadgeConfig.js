import { StrictDict } from '@edx/react-unit-test-utils';

import { useGlobalState } from 'hooks/app';
import {
  stepNames,
  stepStates,
} from 'constants/index';

import messages from './messages';

export const badgeConfig = StrictDict({
  [stepStates.cancelled]: { variant: 'danger', message: messages.cancelled },
  [stepStates.notAvailable]: { variant: 'light', message: messages.notAvailable },
  [stepStates.inProgress]: { variant: 'primary', message: messages.inProgress },
  [stepStates.closed]: { variant: 'danger', message: messages.pastDue },
  [stepStates.needTeam]: { variant: 'warning', message: messages.teamRequired },
  [stepStates.teamAlreadySubmitted]: { variant: 'warning', message: messages.closed },
  [stepStates.waiting]: { variant: 'warning', message: messages.notReady },
  [stepStates.waitingForPeerGrades]: { variant: 'warning', message: messages.waiting },
  [stepNames.done]: { variant: 'success', message: messages.complete },
  staffAfter: StrictDict({
    [stepNames.submission]: { variant: 'primary', message: messages.submitted },
    [stepNames.studentTraining]: { variant: 'primary', message: messages.practiceCompleted },
    [stepNames.self]: { variant: 'primary', message: messages.selfCompleted },
    [stepNames.peer]: { variant: 'primary', message: messages.peerCompleted },
  }),
});

const useBadgeConfig = () => {
  const {
    activeStepName,
    stepState,
    lastStep,
    hasReceivedFinalGrade,
  } = useGlobalState();
  if (hasReceivedFinalGrade) {
    return badgeConfig[stepNames.done];
  }
  if (stepState === stepStates.cancelled) {
    return badgeConfig[stepStates.cancelled];
  }
  if (activeStepName === stepNames.staff) {
    return badgeConfig.staffAfter[lastStep];
  }
  return badgeConfig[stepState];
};

export default useBadgeConfig;
