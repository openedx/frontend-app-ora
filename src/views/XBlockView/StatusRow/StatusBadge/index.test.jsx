import { shallow } from '@edx/react-unit-test-utils';

import StatusBadge from './index';

jest.mock('./useBadgeConfig', () => () => ({
  variant: 'variant',
  message: {
    id: 'message',
    defaultMessage: 'defaultMessage',
  },
}));

describe('StatusBadge', () => {
  it('renders', () => {
    const wrapper = shallow(<StatusBadge />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });
});
