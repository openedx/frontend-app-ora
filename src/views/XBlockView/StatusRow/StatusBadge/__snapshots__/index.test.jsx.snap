// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBadge renders 1`] = `
<Badge
  variant="variant"
>
  defaultMessage
</Badge>
`;
